### DISCUSSION TITLE
Questions about Transcript Security (TLS)

### DISCUSSION THREAD

**Discussion by TaridaGeorge** [LogNet/grpc-spring-boot-starter#275] [General]
![Selection_023](https://user-images.githubusercontent.com/1283959/147885051-76248bec-6923-4893-b132-46f446bcbe65.jpg)

What does the server-cert.pem file should contain? It should have only the root_ca_cert > intermediate_ca_cert or the entire chain root_ca_cert > intermediate_ca_cert > server_cert ?

What file should I use on the client side? 
Does certChain from the `forClient().trustManager(certChain)` should point to the same file as the server property `grpc.security.cert-chain` ?

**Comment by jvmlet** [comment]
[These are](https://github.com/LogNet/grpc-spring-boot-starter/blob/8b05c32dde9c9027344cfd2ec88c66edee44ce91/grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/autoconfigure/GRpcAutoConfiguration.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L129) the grpc config API calls underneath,  please refer to the relevant documentation of `grpc-java`
> Does certChain from the forClient().trustManager(certChain) should point to the same file as the server property grpc.security.cert-chain 

Yes, see how it's configured in tests [here ](https://github.com/LogNet/grpc-spring-boot-starter/blob/8b05c32dde9c9027344cfd2ec88c66edee44ce91/grpc-spring-boot-starter-demo/src/test/java/org/lognet/springboot/grpc/GrpcServerTestBase.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L84)
